eth/tracers/api.go

```diff
@@ -606,24 +607,68 @@ func (api *API) traceBlock(ctx context.Context, block *types.Block, config *Trac
 		return nil, err
 	}
 	defer release()
+	// JS tracers have high overhead. In this case run a parallel
+	// process that generates states in one thread and traces txes
+	// in separate worker threads.
+	if config != nil && config.Tracer != nil && *config.Tracer != "" {
+		t, err := New(*config.Tracer, nil, nil)
+		if err != nil {
+			return nil, err
+		}
+		if _, ok := t.(JSTracer); ok {
+			return api.traceBlockParallel(ctx, block, statedb, config)
+		}
+	}
```

Comment:
This whole thing is a bit wonky, but I guess we can live with it. The alternative would be that they register with some special attribute, and then you can query `IsJs(*config.Tracer)` or `ParallelTracing(*config.Tracer)`. 

I'm fine with it as is, though